test/bufTest.mo

```diff
@@ -92,3 +93,21 @@ func natIterEq(a:I.Iter<Nat>, b:I.Iter<Nat>) : Bool {
   assert (c.toArray().size() == 2);
   assert (c.toVarArray().size() == 2);
 };
+
+// regression test: self-append does not diverge
+{
+  let c = B.Buffer<Nat>(0);
+  let d = B.Buffer<Nat>(0);
+
+  c.add(1); d.add(1);
+  c.add(2); d.add(2);
+  c.add(3); d.add(3);
+
+  Debug.print "append test 1: cloning avoids the issue";
+  d.append(d.clone());
+  Debug.print "append test 2: cloning not necessary";
+  c.append(c);
```

Comment:
Why do we want to allow this? The result is tied closely (maybe too close) to the evaluation order, e.g. what's the result of `c.append(c.append(c)).append(c)`. The `clone()` approach makes this explicit, which is less likely to introduce bugs.

Reply:
Okay, so you dislike option 3.  I can understand why.

So, do you want to choose options 1 or 2 (see my list)?  Or add other design option?

Reply:
I think option 2 makes sense. They can always clone the buffer if they want to self append. I would expect `append` to be associative. Self append breaks that rule.